library/Rules/PostalCode.php

```diff
@@ -81,7 +81,7 @@ class PostalCode extends Regex
         'HT' => '/^(?:HT)*(\d{4})$/',
         'HU' => '/^(\d{4})$/',
         'ID' => '/^(\d{5})$/',
-        'IE' => '/^[A-Z]\d{2}$|^[A-Z]{3}[A-Z]{4}$/',
+        'IE' => '/^[A-Z]\d{2}\s?([A-Z]|\d){4}$/',
```

Comment:
The current validation will not pass with `D6W RY72`.

I was taking a look at this page: https://www.citizensinformation.ie/en/consumer_affairs/telecommunications_and_postal_services/eircode.html#lba536

> The routing key is the first 3 characters of an Eircode. The first character is always a letter, followed by 2 numbers (except for D6W). The letters are not linked to a county or city name, except for postal districts in Dublin, which have had their existing postal codes transferred into a routing key format such as D03, D12 and D22. The same routing key can be shared by several towns and townlands.

Reply:
My mistake. 

 [```^([A-Z]\d{2}|D6W)\s?([A-Z]|\d){4}$```](https://rubular.com/r/WfiGQCquHFsdT1)

This should do instead.